packages/element-web-opendesk-module/src/utils/language.test.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { language } from './language';
+
+describe('language', () => {
+  it('calls getItem with key "mx_local_settings"', () => {
+    jest.spyOn(Storage.prototype, 'getItem');
+    language();
+    expect(localStorage.getItem).toHaveBeenCalledWith('mx_local_settings');
+  });
+
+  it('returns "en" if key does not exist', () => {
+    jest.spyOn(Storage.prototype, 'getItem').mockReturnValue(null);
+    expect(language()).toBe('en');
+  });
+
+  it('returns "de-DE" if settings language is "de"', () => {
+    jest
+      .spyOn(Storage.prototype, 'getItem')
+      .mockReturnValue(JSON.stringify({ language: 'de' }));
+    expect(language()).toBe('de-DE');
+  });
+
+  it('returns "en" if settings language is not supported', () => {
+    jest
+      .spyOn(Storage.prototype, 'getItem')
+      .mockReturnValue(JSON.stringify({ language: 'fr' }));
+    expect(language()).toBe('en');
+  });
+});
```

Comment:
These tests feel odd since they are so split-apart... `mx_local_settings` could (and should) be tested in every test by using `mockImplementation(key => ... );` or just by using `localStorage` which is implemented by jsdom.

Why not do this instead?

```ts
describe('language', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it('returns "en" if key does not exist', () => {
    expect(language()).toBe('en');
  });

  it('returns "de-DE" if settings language is "de"', () => {
    localStorage.setItem('mx_local_settings', '{"language":"de"}');
    expect(language()).toBe('de-DE');
  });

  it('returns "en" if settings language is not supported', () => {
    localStorage.setItem('mx_local_settings', '{"language":"fr"}');
    expect(language()).toBe('en');
  });
});
```